editor/script/editor.js

```diff
@@ -1270,104 +1270,34 @@ function prev() {
 	paintExplorer.ChangeSelection( drawing.id );
 }
 
-function duplicateDrawing() {
-	if (drawing.type == TileType.Tile) {
-
-		//copy drawing data
-		var sourceImageData = renderer.GetImageSource( "TIL_" + drawing.id );
-		var copiedImageData = [];
-		for (f in sourceImageData) {
-			copiedImageData.push([]);
-			for (y in sourceImageData[f]) {
-				copiedImageData[f].push([]);
-				for (x in sourceImageData[f][y]) {
-					copiedImageData[f][y].push( sourceImageData[f][y][x] );
-				}
-			}
-		}
-
-		var tmpIsWall = tile[ drawing.id ].isWall;
-
-		drawing.id = nextTileId();
-
-		console.log("DUPLICATE TILE");		
-		console.log(drawing.id);
-		console.log(copiedImageData);
-
-		makeTile( drawing.id, copiedImageData );
-
-		tile[ drawing.id ].isWall = tmpIsWall;
-
-		paintTool.updateCanvas();
-		refreshGameData();
-
-		tileIndex = Object.keys(tile).length - 1;
-
-		paintTool.reloadDrawing(); //hack for ui consistency
-	}
-	else if(drawing.type == TileType.Avatar || drawing.type == TileType.Sprite) {
-
-		//copy drawing data -- hacky duplication as usual between sprite and tile :(
-		var sourceImageData = renderer.GetImageSource( "SPR_" + drawing.id );
-		var copiedImageData = [];
-		for (f in sourceImageData) {
-			copiedImageData.push([]);
-			for (y in sourceImageData[f]) {
-				copiedImageData[f].push([]);
-				for (x in sourceImageData[f][y]) {
-					copiedImageData[f][y].push( sourceImageData[f][y][x] );
-				}
-			}
-		}
-
-		drawing.id = nextSpriteId();
-
-		console.log("DUPLICATE SPRITE");	
-		console.log(drawing.id);
-		console.log(copiedImageData);
-
-		makeSprite( drawing.id, copiedImageData );
-
-		paintTool.updateCanvas();
-		refreshGameData();
-
-		spriteIndex = Object.keys(sprite).length - 1;
-
-		paintTool.reloadDrawing(); //hack
-	}
-	else if(drawing.type == TileType.Item) {
-
-		//copy drawing data -- hacky duplication as usual between sprite and tile :(
-		var sourceImageData = renderer.GetImageSource( "ITM_" + drawing.id );
-		var copiedImageData = [];
-		for (f in sourceImageData) {
-			copiedImageData.push([]);
-			for (y in sourceImageData[f]) {
-				copiedImageData[f].push([]);
-				for (x in sourceImageData[f][y]) {
-					copiedImageData[f][y].push( sourceImageData[f][y][x] );
-				}
-			}
-		}
-
-		drawing.id = nextItemId();
-
-		console.log("DUPLICATE ITEM");	
-		console.log(drawing.id);
-		console.log(copiedImageData);
+function getIdPrefix(type) {
+    if (type === TileType.Tile) {
+        return "TIL_";
+    } else if (type == TileType.Avatar || type == TileType.Sprite) {
+        return "SPR_";
+    } else if (type == TileType.Item) {
+        return "ITM_";
+    }
+}
 
-		makeItem( drawing.id, copiedImageData );
+function copyDrawingData(sourceDrawingData) {
+    let copiedDrawingData = [];
```

Comment:
I've used `var` everywhere so far; I'd prefer to keep that consistent unless there's specific a reason to use `let` instead